internal/manifests/gateway.go

```diff
@@ -269,54 +279,54 @@ func gatewayConfigOptions(opt Options) gateway.Options {
 	}
 }
 
-func configureGatewayMetricsPKI(podSpec *corev1.PodSpec) error {
+func configureGatewayMetricsPKI(podSpec *corev1.PodSpec, serviceName string) error {
+	var gwIndex int
+	for i, c := range podSpec.Containers {
+		if c.Name == LabelGatewayComponent {
+			gwIndex = i
+			break
+		}
+	}
+
+	secretName := signingServiceSecretName(serviceName)
+	// caFile := path.Join(gateway.LokiGatewayTLSDir, gateway.LokiGatewayCAFile)
+	certFile := path.Join(gateway.LokiGatewayTLSDir, gateway.LokiGatewayCertFile)
+	keyFile := path.Join(gateway.LokiGatewayTLSDir, gateway.LokiGatewayKeyFile)
+
 	secretVolumeSpec := corev1.PodSpec{
 		Volumes: []corev1.Volume{
 			{
-				Name: "tls-secret",
+				Name: tlsMetricsSercetVolume,
 				VolumeSource: corev1.VolumeSource{
 					Secret: &corev1.SecretVolumeSource{
-						SecretName: LabelGatewayComponent,
-					},
-				},
-			},
-			{
-				Name: "tls-configmap",
-				VolumeSource: corev1.VolumeSource{
-					ConfigMap: &corev1.ConfigMapVolumeSource{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: LabelGatewayComponent,
-						},
+						SecretName: secretName,
 					},
 				},
 			},
+			// {
+			//	Name: caConfigMapName,
+			//	VolumeSource: corev1.VolumeSource{
+			//		ConfigMap: &corev1.ConfigMapVolumeSource{
+			//			LocalObjectReference: corev1.LocalObjectReference{
+			//				Name: LabelGatewayComponent,
+			//			},
+			//		},
+			//	},
+			// },
 		},
 	}
 	secretContainerSpec := corev1.Container{
 		VolumeMounts: []corev1.VolumeMount{
 			{
-				Name:      "tls-secret",
-				ReadOnly:  true,
-				MountPath: path.Join(gateway.LokiGatewayTLSDir, "cert"),
-				SubPath:   "cert",
-			},
-			{
-				Name:      "tls-secret",
-				ReadOnly:  true,
-				MountPath: path.Join(gateway.LokiGatewayTLSDir, "key"),
-				SubPath:   "key",
-			},
-			{
-				Name:      "tls-configmap",
+				Name:      tlsMetricsSercetVolume,
 				ReadOnly:  true,
-				MountPath: path.Join(gateway.LokiGatewayTLSDir, "ca"),
-				SubPath:   "ca",
+				MountPath: gateway.LokiGatewayTLSDir,
 			},
 		},
 		Args: []string{
-			fmt.Sprintf("--tls.internal.server.cert-file=%s", path.Join(gateway.LokiGatewayTLSDir, "cert")),
-			fmt.Sprintf("--tls.internal.server.key-file=%s", path.Join(gateway.LokiGatewayTLSDir, "key")),
-			fmt.Sprintf("--tls.healthchecks.server-ca-file=%s", path.Join(gateway.LokiGatewayTLSDir, "ca")),
+			fmt.Sprintf("--tls.internal.server.cert-file=%s", certFile),
+			fmt.Sprintf("--tls.internal.server.key-file=%s", keyFile),
+			// fmt.Sprintf("--tls.healthchecks.server-ca-file=%s", caFile),
```

Comment:
@periklis Any reason why we have commented out things related to `server-ca-file` everywhere?

Reply:
@sasagarw The CA file is something we need to implement for TLS metrics. This requires a separate controller to reconcile for the TrustedCABundle in OCP as we similarly do in EO and CLO for Kibana. Currently the pods don't start with these options.